mongo/integration/mtest/mongotest.go

```diff
@@ -666,13 +666,26 @@ func verifyTopologyConstraints(topologies []TopologyKind) error {
 	return fmt.Errorf("topology kind %q does not match any of the required kinds %q", testContext.topoKind, topologies)
 }
 
+func verifyServerParametersConstraints(serverParameters map[string]bson.RawValue) error {
+	for param, expectedValue := range serverParameters {
+		actualValue := testContext.serverParameters.Lookup(param)
```

Comment:
Because we're no longer relying on these being boolean params, the error message we emit should be clearer and should also distinguish between the server not supporting a parameter and the server reporting the wrong value for a parameter. We can use `LookupErr` for to separate the cases and emit better messages:

```
actual, err := testContext.serverParameters.LookupErr(param)
if err != nil {
    return fmt.Errorf("server does not support parameter %q", ...)
}
if !expectedValue.Equal(actualValue) {
    return fmt.Errorf("mismatched values for server parameter %q; expected %s, got %s", ...)
}
```

The code is just meant to give you an idea of the control flow I'm thinking about. Feel free to make tweaks as needed.

Reply:
Great idea! Just added in almost exactly that code haha.